src/applications/gi/utils/helpers.js

```diff
@@ -88,3 +90,56 @@ export const formatDollarAmount = value => {
   const output = value != null ? removeNonNumberCharacters(value) : 0;
   return formatCurrency(output);
 };
+
+export const isMobileView = () => window.innerWidth <= SMALL_SCREEN_WIDTH;
+
+export const handleScrollOnInputFocus = fieldId => {
+  const field = document.getElementById(fieldId);
+  if (isMobileView()) {
+    // prod flag for bah-8821 EYB changes
+    if (environment.isProduction()) {
+      field.scrollIntoView();
+    } else {
+      field.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+  }
+};
+
+export const scrollUpIfOverlap = (fieldId1, fieldId2, scrollableFieldId) => {
+  const field1 = document.getElementById(fieldId1);
+  const field2 = document.getElementById(fieldId2);
+  if (field1 && field2) {
+    const fieldRect1 = field1.getBoundingClientRect();
+    const fieldRect2 = field2.getBoundingClientRect();
+    const hasOverLap = !(
+      fieldRect1.right < fieldRect2.left ||
+      fieldRect1.left > fieldRect2.right ||
+      fieldRect1.bottom < fieldRect2.top ||
+      fieldRect1.top > fieldRect2.bottom
+    );
+    if (hasOverLap === true) {
+      const scrollableField =
+        document.getElementById(scrollableFieldId) || window;
+      const scrollUpBy = fieldRect1.bottom - fieldRect2.top + 2;
+      scrollableField.scrollBy({
+        top: scrollUpBy,
+        left: 0,
+        behavior: 'smooth',
+      });
+    }
+  }
+};
+
+export const handleInputFocusWithPotentialOverLap = (
+  fieldId1,
+  fieldId2,
+  scrollableFieldId,
+) => {
+  if (isMobileView()) {
+    handleScrollOnInputFocus(fieldId1);
+    // prod flag for bah-8821 EYB changes
+    if (!environment.isProduction()) {
```

Comment:
prod flag for `handleInputFocusWithPotentialOverLap` should be in `src/applications/gi/containers/SearchPage.jsx` or `src/applications/gi/components/search/InstitutionSearchForm.jsx` as only use of it that needs a prod flag is in those classes, the other usage of the function is behind the `gibctEstimateYourBenefits` feature flag the wraps `src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx`

Reply:
Removing the 2 prod flags in `src/applications/gi/utils/helpers.js` would require more work when we have to remove them down the road. Is there a reason to do this other than that it would remove the two helper methods I created in `src/applications/gi/tests/e2e/gibct-helpers.js`?

Reply:
Removing the prod flag for `handleInputFocusWithPotentialOverLap` will not create extra work as one usage of this function is behind the feature flag `gibctEstimateYourBenefits` that wraps `src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx`  and the other is in `src/applications/gi/containers/SearchPage.jsx` and therefore can be moved there.

There is not a need to have something behind both a feature flag and a prod flag.

The goal is avoid changing the e2e tests and creating duplicate helper functions.

Reply:
Okay, the changes are in.